core/src/test/java/org/testcontainers/containers/ReusabilityUnitTests.java

```diff
@@ -0,0 +1,200 @@
+            .withNetworkMode("none") // to disable the port forwarding
+            .withStartupCheckStrategy(startupCheckStrategy)
+            .waitingFor(waitStrategy)
+            .withReuse(true);
+
+        {
+            container.dockerClient = client;
+        }
+
+        @Test
+        public void shouldStartIfListReturnsEmpty() {
+            String containerId = randomContainerId();
+            when(client.createContainerCmd(any())).then(createContainerAnswer(containerId));
+            when(client.listContainersCmd()).then(listContainersAnswer());
+            when(client.startContainerCmd(containerId)).then(startContainerAnswer());
+            when(client.inspectContainerCmd(containerId)).then(inspectContainerAnswer());
+
+            container.start();
+
+            Mockito.verify(client, Mockito.atLeastOnce()).startContainerCmd(containerId);
+        }
+
+        @Test
+        public void shouldReuseIfListReturnsID() {
+            // TODO mock TestcontainersConfiguration
+            Assume.assumeTrue("supports reuse", TestcontainersConfiguration.getInstance().environmentSupportsReuse());
+            when(client.createContainerCmd(any())).then(createContainerAnswer(randomContainerId()));
+            String existingContainerId = randomContainerId();
+            when(client.listContainersCmd()).then(listContainersAnswer(existingContainerId));
+            when(client.inspectContainerCmd(existingContainerId)).then(inspectContainerAnswer());
+
+            container.start();
+
+            Mockito.verify(client, Mockito.never()).startContainerCmd(existingContainerId);
+        }
+
+        private String randomContainerId() {
+            return UUID.randomUUID().toString();
+        }
+
+        private Answer<ListContainersCmd> listContainersAnswer(String... ids) {
+            return invocation -> {
+                ListContainersCmd.Exec exec = command -> {
+                    return new ObjectMapper().convertValue(
+                        Stream.of(ids)
+                            .map(id -> Collections.singletonMap("Id", id))
+                            .collect(Collectors.toList()),
+                        new TypeReference<List<Container>>() {}
+                    );
+                };
+                return new ListContainersCmdImpl(exec);
+            };
+        }
+
+        private Answer<CreateContainerCmd> createContainerAnswer(String containerId) {
+            return invocation -> {
+                CreateContainerCmd.Exec exec = command -> {
+                    CreateContainerResponse response = new CreateContainerResponse();
+                    response.setId(containerId);
+                    return response;
+                };
+                return new CreateContainerCmdImpl(exec, null, "image:latest");
+            };
+        }
+
+        private Answer<StartContainerCmd> startContainerAnswer() {
+            return invocation -> {
+                StartContainerCmd.Exec exec = command -> {
```

Comment:
```suggestion
                StartContainerCmd.Exec exec = command -> null
```
multiline suggest still not possible, but you get the idea?

Reply:
ah, okay. I wanted to keep it consistent with the rest of the `exec` mocks and make them look method-like.

WDYT?

Reply:
yes, sounds good. Just got triggered by the IntelliJ marking.